glnx-fdio.h

```diff
@@ -167,6 +167,37 @@ int glnx_renameat2_exchange (int olddirfd, const char *oldpath,
                              int newdirfd, const char *newpath);
 
 /**
+ * glnx_fallocate:
+ * @fd: File descriptor
+ * @size: Size
+ * @error: Error
+ *
+ * Wrapper for Linux fallocate().  Explicitly ignores a @size of zero.
+ * Use this instead of posix_fallocate(), since the glibc fallback
+ * is bad: https://sourceware.org/bugzilla/show_bug.cgi?id=18515
+ */
+static inline gboolean
+glnx_fallocate (int      fd,
```

Comment:
Hmm, I'd name this something like `glnx_try_fallocate` or `glnx_maybe_fallocate` instead.

Otherwise, LGTM!